ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/model/SchemaNode.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.confluent.ksql.serde.PhysicalSchema;
+import io.confluent.ksql.serde.SerdeOption;
+import java.util.Objects;
+import java.util.Set;
+
+public class SchemaNode {
+
+  private String logicalSchema;
+  private Set<SerdeOption> serdeOptions;
+
+  public SchemaNode(
+      @JsonProperty("schema") final String logicalSchema,
+      @JsonProperty("serdeOptions") final Set<SerdeOption> serdeOptions
+  ) {
+    this.logicalSchema = Objects.requireNonNull(logicalSchema, "logicalSchema");
+    this.serdeOptions = Objects.requireNonNull(serdeOptions, "serdeOptions");
+  }
+
+  @JsonProperty("schema")
+  public String getLogicalSchema() {
+    return logicalSchema;
+  }
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
```

Comment:
serdeOptions is meant to always be in the serialization, even if the set is empty. Empty set has meaning for verification: it means to verify that the set of serde options in the source is empty. If someone introduces a change in the code that causes one of the sources that previously had no custom serde options to suddenly have one, a QTT should fail.

This fits with your suggestion to serialize PhysicalSchema directly rather than introducing SchemaNode, so I'll go with that.